wellcomecollection/wellcomecollection.org
# Compare event target with currentTarget

For #11012

## What does this change?

We had previously used `event.stopPropagation()` to stop the clicking of an inner TreeItem from bubbling up to any outer TreeItems, in order to e.g. prevent the tree view from closing when you clicked on an inner item.

But this prevented a GTM trigger that we have set on download links from firing.

So instead, we now check that we only fire the click event on the _original_ target of the click by comparing `target` (original) with `currentTarget` (whatever is currently being evaluated in the bubbling phase).

## How to test

1. Turn on the Display born digital files [toggle](https://dash.wellcomecollection.org/toggles/)
2. Turn on the GA debug chrome extension (or use the GTM preview)
3. Visit a work with born digital items (e.g. http://localhost:3000/works/ar85nv5f)
4. Confirm clicking on an item with subitems will toggle it open/closed
5. Confirm clicking on a subitem will not close the parent container
6. Confirm clicking on a 'download' link in the table will fire the `download_table_link` GTM trigger

## How can we measure success?

We have analytics for numbers of born digital downloads

## Have we considered potential risks?

Can't think of any
